### PR TITLE
Add e2fsprogs to xmake repo

### PR DESCRIPTION
**Pull request by zhangxp1998**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

File: packages/e/e2fsprogs/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,30 @@
+
+
```
Comment (waruqi) [inline packages/e/e2fsprogs/xmake.lua:2]: please remove empty lines

**Comment by zhangxp1998** [comment]
Done

File: packages/e/e2fsprogs/xmake.lua (hunk 1 above)
Comment (zhangxp1998) [reply packages/e/e2fsprogs/xmake.lua]: Removed

File: packages/e/e2fsprogs/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    add_versions("1.46.4", "c011bf3bf4ae5efe9fa2b0e9b0da0c14ef4b79c6143c1ae6d9f027931ec7abe1")
+
+    if is_plat("linux") then
+        add_extsources("apt::e2fsprogs")
+    elseif is_plat("macosx") then
+        add_extsources("brew::e2fsprogs")
+    end
+
+    on_install(function (package)
```
Comment (waruqi) [inline packages/e/e2fsprogs/xmake.lua:16]: ios, windows ci fails. you can just add supported platforms here.

for example:

```
on_install("linux", "macosx", function(package)
end
```

File: packages/e/e2fsprogs/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+		import("package.tools.autoconf")
+		-- Enforce MKDIR_P to work around a configure bug
+		-- see https://github.com/Homebrew/homebrew-core/pull/35339
+		autoconf.build(package, {"MKDIR_P=mkdir -p", "--disable-e2initrd-helper"})
+		-- make V=1 will fail for e2fsprogs, for easons unknown
+		-- So call make manually to ensure V=1 is not specified
+		autoconf.make(package, {"install"})
```
Comment (waruqi) [inline packages/e/e2fsprogs/xmake.lua:23]: please convert tab to 4 spaces indent

File: packages/e/e2fsprogs/xmake.lua (hunk 2 above)
Comment (zhangxp1998) [reply packages/e/e2fsprogs/xmake.lua]: Done

File: packages/e/e2fsprogs/xmake.lua (hunk 3 above)
Comment (zhangxp1998) [reply packages/e/e2fsprogs/xmake.lua]: Done

File: packages/e/e2fsprogs/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        add_extsources("apt::e2fsprogs")
+    elseif is_plat("macosx") then
+        add_extsources("brew::e2fsprogs")
+    end
+
+    on_install("linux", "macosx", function (package)
+        import("package.tools.autoconf")
+        local configs = {"MKDIR_P=mkdir -p", "--disable-e2initrd-helper"}
+        if package:config("shared") then
+            if is_plat("linux") then
```
Comment (waruqi) [inline packages/e/e2fsprogs/xmake.lua:20]: please use `package:is_plat`

File: packages/e/e2fsprogs/xmake.lua [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+                table.insert(configs, "--enable-elf-shlibs")
+            elseif is_plat("macosx") then
```
Comment (waruqi) [inline packages/e/e2fsprogs/xmake.lua:22]: and here

Comment (zhangxp1998) [reply packages/e/e2fsprogs/xmake.lua]: done

File: packages/e/e2fsprogs/xmake.lua (hunk 4 above)
Comment (zhangxp1998) [reply packages/e/e2fsprogs/xmake.lua]: done

**Comment by waruqi** [comment]
Linux ci failed. It seems that some files will still be installed in the system /lib directory. It seems that xmake just set the --prefix parameter to no effect. Maybe there are other parameters to modify the installation directory?

**Comment by waruqi** [comment]
> 	MKDIR_P /etc/cron.d
> /usr/bin/install: cannot remove '/lib/udev/rules.d/96-e2scrub.rules': Permission denied

maybe you need set `--libdir=` and `--sysconfdir=` to `package:installdir("lib")` and `package:installdir("etc")`

**Comment by waruqi** [comment]
> rw_bitmaps.c:(.text+0x16c3): undefined reference to `pthread_create'

you need add 

```lua
if is_plat("linux") then
    add_syslinks("pthread")
end
```

File: packages/e/e2fsprogs/xmake.lua [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        add_extsources("apt::e2fsprogs")
+    elseif is_plat("macosx") then
+        add_extsources("brew::e2fsprogs")
+    end
+    if is_plat("linux") then
+        add_syslinks("pthread")
+    end
+
+    on_install("linux", "macosx", function (package)
+        os.vrunv("rm", "-rf", "scrub")
```
Comment (waruqi) [inline packages/e/e2fsprogs/xmake.lua:20]: `os.rm("scrub")`

File: packages/e/e2fsprogs/xmake.lua [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        import("package.tools.autoconf")
```
Comment (waruqi) [inline packages/e/e2fsprogs/xmake.lua:21]: please move to the first line

**Comment by waruqi** [comment]
> alloc_stats.c:(.text+0xd8): undefined reference to `com_err'

add_syslinks("com_err") for linux?

**Comment by waruqi** [comment]
thanks!

**Pull request merged**